### DISCUSSION TITLE
gradient color on `QPalette::highlight` role in lxqt-qtplugin

### DISCUSSION THREAD

**Discussion by hamzehnasajpour** [lxqt/lxqt#2084] [Q&A]
I wanted to have gradient style on `highlight` role in `QPalette`, so I created a `QLinearGradient` object and set it as a `QBrush` on `QPalette::Highlight`, but it doesn't doesn't work.

```
QLinearGradient gradient(QPointF(1, 1), QPointF(0,1));
gradient.setColorAt(0, "#c26ad5");
gradient.setColorAt(1, "#452db3");
gradient.setCoordinateMode(QGradient::StretchToDeviceMode);
QBrush brush(gradient);
LXQtPalette_->setBrush(QPalette::Highlight, gradient);
...
```

Is there anything that I missed? The point is, as a test I set it on `QPalette::Window` and it works.

**Comment by tsujan** [comment]
`QPalette` doesn't style anything; `QStyle` does, through widget styles (aka. style plugins). A widget style can get colors of `QPalette` and do whatever it wants to do with them or, as [Qt doc](https://doc.qt.io/qt-5/qpalette.html#details) says, it can ignore them completely. LXQt provides palettes only for widget styles that use them, like Fusion. How they use them is determined by their codes.

**Reply by hamzehnasajpour** [reply]
@tsujan Maybe my description was not good enough. I just want to create a linear gradient color and set it as `Highlight` role in `QPalette`. Of-course I want to know can I set this gradient on that role in `LXQtPalette_`? Like this [example](https://wiki.qt.io/Creating_a_styled_Gradient_Button_Bar#Enter_QBrush.2C_QLinearGradient_and_QPalette) but on the `LXQtPalette_` (above code snippet)

**Reply by tsujan** [reply]
> Maybe my description was not good enough.

It was.

> I want to know can I set this gradient on that role in LXQtPalette_?

You can. And, as you've seen yourself, it guarantees nothing, for the reason I mentioned above.
